log/logClient_test.go

```diff
@@ -39,22 +66,47 @@ func TestLogClient(t *testing.T) {
 		ContainerID:    "12345678901234567890",
 		ContainerName:  "los-polos",
 		ContainerImage: "evergreen",
+		Type:           "MatchedPolicy",
 	}
-	eventChan = make(chan interface{}, maxEvents)
+	eventChan = make(chan []byte, maxEvents)
 	var o = Options{
 		EventChan: eventChan,
 	}
+
+	tel, err := json.Marshal(res)
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+
+	// Handle Telemetry Events in Generic Way
 	for i := 0; i < maxEvents; i++ {
-		err := watchAlertsHelper(&res, o)
-		if err != nil {
-			t.Errorf("watchAlertsHelper failed\n")
+		WatchTelemetryHelper(tel, "Alert", o)
+	}
+	done = make(chan bool, 1)
+	go genericWaitOnEvent(maxEvents)
+	// Check for timeouts
+	select {
+	case <-done:
+		if gotAlerts < maxEvents {
+			t.Errorf("did not receive all the events")
 		}
+	case <-time.After(100 * time.Millisecond):
+		t.Errorf("timed out")
 	}
-	go waitOnEvent(maxEvents)
-	for i := 0; i < 10 && gotAlerts < maxEvents; i++ {
-		time.Sleep(10 * time.Millisecond)
+
+	// Handle Alert Events
+	for i := 0; i < maxEvents; i++ {
+		WatchTelemetryHelper(tel, "Alert", o)
 	}
-	if gotAlerts < maxEvents {
-		t.Errorf("did not receive all the events")
+	gotAlerts = 0
+	go waitOnAlert(maxEvents)
```

Comment:
we can remove `waitOnAlert()` and this code section ... doesn't serve any new purpose since there is `waitOnGenericEvent()` now.